lib/components/message_input.dart

```diff
@@ -20,6 +23,89 @@ class _MessageInputWidgetState extends State<MessageInputWidget> {
   final _chatInputFocusNode = FocusNode();
   var _isEmotePickerVisible = false;
 
+  OverlayEntry? entry;
+
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  bool startsWithPossibleCommands(String text) {
+    if (text == "" || text.isEmpty) {
+      return false;
+    }
+    for (final mode in ChatMode.chatModes) {
+      if (mode.title.startsWith(text)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  void hideOverlay() {
+    entry?.remove();
+    entry = null;
+  }
+
+  void showOverlay(String text) {
+    // remove existing overlay, bc user can contiunously type a prefix string that matches a command
+    hideOverlay();
+
+    final overlay = Overlay.of(context)!;
+
+    // the renderbox of this widget
+    final renderBox = context.findRenderObject() as RenderBox;
+    final size = renderBox.size;
+
+    final offset = renderBox.localToGlobal(Offset.zero);
+    final lst =
+        ChatMode.chatModes.where((element) => element.title.startsWith(text));
+
+    // None to show
+    if (lst.isEmpty) {
+      hideOverlay();
+      return;
+    }
+
+    final lstSize = lst.length;
+    const listTileSize = 75; // the roughly size of a listTile
+    final shiftUp = min(lstSize * listTileSize, 300);
+
+    entry = OverlayEntry(builder: (context) {
```

Comment:
Can we make the state whether or not the widget is visible and render this in the `build()` function? Leaving the Widget construction outside `build()` can be a bit dangerous.

Reply:
I'm not sure if I can move the `Overlay` widget into the `build()`. 
Rendering a `Overlay` seems to require insertion and removal for delete.